ci-operator/step-registry/optional-operators/subscribe/optional-operators-subscribe-commands.sh

```diff
@@ -93,28 +112,65 @@ spec:
 EOF
 )
 
-echo "subscription name is \"$SUB\""
-echo "waiting for csv to become ready..."
+echo "Subscription name is \"$SUB\""
+echo "Waiting for ClusterServiceVersion to become ready..."
 
 for _ in $(seq 1 30); do
     CSV=$(oc -n "$OO_INSTALL_NAMESPACE" get subscription "$SUB" -o jsonpath='{.status.installedCSV}' || true)
     if [[ -n "$CSV" ]]; then
-	if [[ "$(oc -n "$OO_INSTALL_NAMESPACE" get csv "$CSV" -o jsonpath='{.status.phase}')" == "Succeeded" ]]; then
-	    echo "csv \"$CSV\" ready"
-	    exit 0
-	fi
+        if [[ "$(oc -n "$OO_INSTALL_NAMESPACE" get csv "$CSV" -o jsonpath='{.status.phase}')" == "Succeeded" ]]; then
+            echo "ClusterServiceVersion \"$CSV\" ready"
+            exit 0
+        fi
     fi
     sleep 10
 done
 
-echo "timed out waiting for csv to become ready" 1>&2
-oc get namespace "$OO_INSTALL_NAMESPACE" -o yaml 1>&2
-oc get -n "$OO_INSTALL_NAMESPACE" operatorgroup "$OPERATORGROUP" -o yaml 1>&2
-oc get -n "$OO_INSTALL_NAMESPACE" catalogsource "$CATSRC" -o yaml 1>&2
-oc get -n "$OO_INSTALL_NAMESPACE" subscription "$SUB" -o yaml 1>&2
+echo "Timed out waiting for csv to become ready" 1>&2
+
+NS_ART="$ARTIFACT_DIR/ns-$OO_INSTALL_NAMESPACE.yaml"
```

Comment:
Is this reusable on teardown/cleanup?

Reply:
I don't think teardown or cleanup needs anything like this; either the installation succeeds and the resources are fine, or it fails and they are captured. Can you elaborate on the question?

Reply:
Not much to elaborate on -- if these objects contain status, etc, that could be useful in debugging something post-facto it could be something nice to capture. Not sure if it's useful or not.

Reply:
Oh, I see. Not sure: it's definitely doable but we would need to pass some identifiers (installation namespace etc) via `SHARED_DIR` so that the teardown can pick them up.

I wasn't expecting tests that follow this step to mess with catalogsources, subscriptions etc, so I didn't think a separate teardown step is necessary, but that's just an assumption.